In [1]:
import pandas as pd
import numpy as np

In [3]:
# Open csv file
df = pd.read_csv("googleplaystore.csv")
df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [4]:
df.shape

(10841, 13)

In [5]:
tot = df.isnull().sum().sort_values(ascending = False)
pd.concat([tot], axis = 1, keys = ["Total"])

Total
Rating           1474
Current Ver         8
Android Ver         3
Content Rating      1
Type                1
Last Updated        0
Genres              0
Price               0
Installs            0
Size                0
Reviews             0
Category            0
App                 0

In [5]:
df.dropna(inplace = True)
df.shape

(9360, 13)

In [6]:
print(len(df["Category"].unique()),"Categories", "\n")
print(df["Category"].unique())

33 Categories 

['ART_AND_DESIGN' 'AUTO_AND_VEHICLES' 'BEAUTY' 'BOOKS_AND_REFERENCE'
 'BUSINESS' 'COMICS' 'COMMUNICATION' 'DATING' 'EDUCATION' 'ENTERTAINMENT'
 'EVENTS' 'FINANCE' 'FOOD_AND_DRINK' 'HEALTH_AND_FITNESS' 'HOUSE_AND_HOME'
 'LIBRARIES_AND_DEMO' 'LIFESTYLE' 'GAME' 'FAMILY' 'MEDICAL' 'SOCIAL'
 'SHOPPING' 'PHOTOGRAPHY' 'SPORTS' 'TRAVEL_AND_LOCAL' 'TOOLS'
 'PERSONALIZATION' 'PRODUCTIVITY' 'PARENTING' 'WEATHER' 'VIDEO_PLAYERS'
 'NEWS_AND_MAGAZINES' 'MAPS_AND_NAVIGATION']


In [7]:
print(len(df["Genres"].unique()),"Genres", "\n")
print(df["Genres"].unique())

115 Genres 

['Art & Design' 'Art & Design;Pretend Play' 'Art & Design;Creativity'
 'Auto & Vehicles' 'Beauty' 'Books & Reference' 'Business' 'Comics'
 'Comics;Creativity' 'Communication' 'Dating' 'Education;Education'
 'Education' 'Education;Creativity' 'Education;Music & Video'
 'Education;Action & Adventure' 'Education;Pretend Play'
 'Education;Brain Games' 'Entertainment' 'Entertainment;Music & Video'
 'Entertainment;Brain Games' 'Entertainment;Creativity' 'Events' 'Finance'
 'Food & Drink' 'Health & Fitness' 'House & Home' 'Libraries & Demo'
 'Lifestyle' 'Lifestyle;Pretend Play' 'Adventure;Action & Adventure'
 'Arcade' 'Casual' 'Card' 'Casual;Pretend Play' 'Action' 'Strategy'
 'Puzzle' 'Sports' 'Music' 'Word' 'Racing' 'Casual;Creativity'
 'Casual;Action & Adventure' 'Simulation' 'Adventure' 'Board' 'Trivia'
 'Role Playing' 'Simulation;Education' 'Action;Action & Adventure'
 'Casual;Brain Games' 'Simulation;Action & Adventure'
 'Educational;Creativity' 'Puzzle;Brain Games' 'Educati

## Find Most Expensive Apps

In [8]:
df["Type"].unique()

array(['Free', 'Paid'], dtype=object)

In [9]:
df_paid = df[df["Type"] == "Paid"]
df_paid.head()

App       Category  Rating  \
234  TurboScan: scan documents and receipts in PDF       BUSINESS     4.7   
235                 Tiny Scanner Pro: PDF Doc Scan       BUSINESS     4.8   
290  TurboScan: scan documents and receipts in PDF       BUSINESS     4.7   
291                 Tiny Scanner Pro: PDF Doc Scan       BUSINESS     4.8   
427                             Puffin Browser Pro  COMMUNICATION     4.0   

    Reviews                Size  Installs  Type  Price Content Rating  \
234   11442                6.8M  100,000+  Paid  $4.99       Everyone   
235   10295                 39M  100,000+  Paid  $4.99       Everyone   
290   11442                6.8M  100,000+  Paid  $4.99       Everyone   
291   10295                 39M  100,000+  Paid  $4.99       Everyone   
427   18247  Varies with device  100,000+  Paid  $3.99       Everyone   

            Genres    Last Updated  Current Ver Android Ver  
234       Business  March 25, 2018        1.5.2  4.0 and up  
235       Business  April 11, 2017        3.4.6  3.0 and up  
290       Business  March 25, 2018        1.5.2  4.0 and up  
291       Business  April 11, 2017        3.4.6  3.0 and up  
427  Communication    July 5, 2018  7.5.3.20547  4.1 and up

In [10]:
df_paid.Price = df_paid.Price.apply(lambda x: x.replace('$',''))
df_paid["Price"] = df_paid["Price"].apply(lambda x: float(x))

/opt/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py:5303: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value
/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [11]:
df_paid["Price"].max()

400.0

In [12]:
df_paid.sort_values("Price", ascending = False).head(10)

App   Category  Rating Reviews  Size  \
4367        I'm Rich - Trump Edition  LIFESTYLE     3.6     275  7.3M   
5356               I Am Rich Premium    FINANCE     4.1    1867  4.7M   
4362                      💎 I'm rich  LIFESTYLE     3.8     718   26M   
5364  I am rich (Most expensive app)    FINANCE     4.1     129  2.7M   
4197          most expensive app (H)     FAMILY     4.3       6  1.5M   
5362                   I Am Rich Pro     FAMILY     4.4     201  2.7M   
5373              I AM RICH PRO PLUS    FINANCE     4.0      36   41M   
5359              I am rich(premium)    FINANCE     3.5     472  965k   
5358                      I am Rich!    FINANCE     3.8      93   22M   
5351                       I am rich  LIFESTYLE     3.8    3547  1.8M   

      Installs  Type   Price Content Rating         Genres       Last Updated  \
4367   10,000+  Paid  400.00       Everyone      Lifestyle        May 3, 2018   
5356   50,000+  Paid  399.99       Everyone        Finance  November 12, 2017   
4362   10,000+  Paid  399.99       Everyone      Lifestyle     March 11, 2018   
5364    1,000+  Paid  399.99           Teen        Finance   December 6, 2017   
4197      100+  Paid  399.99       Everyone  Entertainment      July 16, 2018   
5362    5,000+  Paid  399.99       Everyone  Entertainment       May 30, 2017   
5373    1,000+  Paid  399.99       Everyone        Finance      June 25, 2018   
5359    5,000+  Paid  399.99       Everyone        Finance        May 1, 2017   
5358    1,000+  Paid  399.99       Everyone        Finance  December 11, 2017   
5351  100,000+  Paid  399.99       Everyone      Lifestyle   January 12, 2018   

     Current Ver   Android Ver  
4367       1.0.1    4.1 and up  
5356         1.6    4.0 and up  
4362       1.0.0    4.4 and up  
5364           2  4.0.3 and up  
4197         1.0    7.0 and up  
5362        1.54    1.6 and up  
5373       1.0.2    4.1 and up  
5359         3.4    4.4 and up  
5358         1.0    4.1 and up  
5351         2.0  4.0.3 and up

## Find Least Expensive Apps

In [13]:
df_paid["Price"].min()

0.99

In [14]:
df_paid.sort_values("Price", ascending = True).head(10)

App         Category  Rating Reviews  Size  \
5777   F05WatchFace for Android Wear     PRODUCTIVITY     3.9     127  5.9M   
4003         Neon Glow C - Icon Pack  PERSONALIZATION     4.8     557   74M   
10035                     Deus Ex GO           FAMILY     4.5    9699   23M   
4034                   Hitman Sniper             GAME     4.6  408292   29M   
4047              Koi Live Wallpaper  PERSONALIZATION     4.6   14026   12M   
8993              DW Contacts widget            TOOLS     4.3      56  2.0M   
5557          Nightenfell: Shared AR             GAME     4.5      20   65M   
4169                              Go  PERSONALIZATION     4.6      89   11M   
10039  Device Info Ex Live Wallpaper  PERSONALIZATION     4.1    2645  1.1M   
6584              iBP Blood Pressure          MEDICAL     4.4     578  704k   

          Installs  Type  Price Content Rating           Genres  \
5777        1,000+  Paid   0.99       Everyone     Productivity   
4003       10,000+  Paid   0.99       Everyone  Personalization   
10035     100,000+  Paid   0.99           Teen           Puzzle   
4034   10,000,000+  Paid   0.99     Mature 17+           Action   
4047      100,000+  Paid   0.99       Everyone  Personalization   
8993        1,000+  Paid   0.99       Everyone            Tools   
5557        1,000+  Paid   0.99   Everyone 10+           Action   
4169        5,000+  Paid   0.99       Everyone  Personalization   
10039      50,000+  Paid   0.99       Everyone  Personalization   
6584       10,000+  Paid   0.99       Everyone          Medical   

            Last Updated Current Ver   Android Ver  
5777   December 24, 2017       7.0.1    4.3 and up  
4003       July 27, 2018       4.5.0  4.0.3 and up  
10035      July 23, 2018  2.1.111374    4.2 and up  
4034       July 12, 2018  1.7.110758    4.1 and up  
4047       June 24, 2014         1.9    2.2 and up  
8993        June 5, 2014     1.1.0.0    2.1 and up  
5557      March 27, 2018       1.0.0    7.0 and up  
4169    October 21, 2017       1.6.1  4.0.3 and up  
10039  December 20, 2015      11.0.0    2.1 and up  
6584   November 30, 2014       7.0.1    2.2 and up

## Count Number of Apps in each Category

In [15]:
df_cat = df.Category.value_counts().reset_index().rename(columns = {"index": "Category", "Category": "Count"})
df_cat

Category  Count
0                FAMILY   1746
1                  GAME   1097
2                 TOOLS    733
3          PRODUCTIVITY    351
4               MEDICAL    350
5         COMMUNICATION    328
6               FINANCE    323
7                SPORTS    319
8           PHOTOGRAPHY    317
9             LIFESTYLE    314
10      PERSONALIZATION    312
11             BUSINESS    303
12   HEALTH_AND_FITNESS    297
13               SOCIAL    259
14             SHOPPING    238
15   NEWS_AND_MAGAZINES    233
16     TRAVEL_AND_LOCAL    226
17               DATING    195
18  BOOKS_AND_REFERENCE    178
19        VIDEO_PLAYERS    160
20            EDUCATION    155
21        ENTERTAINMENT    149
22  MAPS_AND_NAVIGATION    124
23       FOOD_AND_DRINK    109
24       HOUSE_AND_HOME     76
25              WEATHER     75
26    AUTO_AND_VEHICLES     73
27   LIBRARIES_AND_DEMO     64
28       ART_AND_DESIGN     61
29               COMICS     58
30            PARENTING     50
31               EVENTS     45
32               BEAUTY     42

## Find Most Downloaded Apps

In [16]:
df.Installs.replace("[\+,]","", regex = True, inplace = True)

In [17]:
df.sort_values("Installs", ascending = False).head(10)

App       Category  Rating   Reviews  \
4676                Viber Messenger  COMMUNICATION     4.3  11335481   
1908                 My Talking Tom           GAME     4.5  14892469   
1842               Candy Crush Saga           GAME     4.4  22429716   
2546                  Facebook Lite         SOCIAL     4.3   8606259   
1885                            Pou           GAME     4.3  10486018   
1886                   Temple Run 2           GAME     4.3   8119151   
465   imo free video calls and chat  COMMUNICATION     4.3   4785988   
347   imo free video calls and chat  COMMUNICATION     4.3   4785892   
467                 Viber Messenger  COMMUNICATION     4.3  11335255   
1729                   Temple Run 2           GAME     4.3   8118937   

                    Size   Installs  Type Price Content Rating         Genres  \
4676  Varies with device  500000000  Free     0       Everyone  Communication   
1908  Varies with device  500000000  Free     0       Everyone         Casual   
1842                 74M  500000000  Free     0       Everyone         Casual   
2546  Varies with device  500000000  Free     0           Teen         Social   
1885                 24M  500000000  Free     0       Everyone         Casual   
1886                 62M  500000000  Free     0       Everyone         Action   
465                  11M  500000000  Free     0       Everyone  Communication   
347                  11M  500000000  Free     0       Everyone  Communication   
467   Varies with device  500000000  Free     0       Everyone  Communication   
1729                 62M  500000000  Free     0       Everyone         Action   

        Last Updated         Current Ver         Android Ver  
4676   July 18, 2018  Varies with device  Varies with device  
1908   July 19, 2018           4.8.0.132          4.1 and up  
1842    July 5, 2018           1.129.0.2          4.1 and up  
2546  August 1, 2018  Varies with device  Varies with device  
1885    May 25, 2018              1.4.77          4.0 and up  
1886    July 5, 2018              1.49.1          4.0 and up  
465     June 8, 2018    9.8.000000010501          4.0 and up  
347     June 8, 2018    9.8.000000010501          4.0 and up  
467    July 18, 2018  Varies with device  Varies with device  
1729    July 5, 2018              1.49.1          4.0 and up

## Calculate Average Rating in each Category

In [18]:
df.groupby("Category").mean()

Rating
Category                     
ART_AND_DESIGN       4.377049
AUTO_AND_VEHICLES    4.190411
BEAUTY               4.278571
BOOKS_AND_REFERENCE  4.346067
BUSINESS             4.121452
COMICS               4.155172
COMMUNICATION        4.158537
DATING               3.970769
EDUCATION            4.389032
ENTERTAINMENT        4.126174
EVENTS               4.435556
FAMILY               4.192383
FINANCE              4.131889
FOOD_AND_DRINK       4.166972
GAME                 4.286326
HEALTH_AND_FITNESS   4.277104
HOUSE_AND_HOME       4.197368
LIBRARIES_AND_DEMO   4.179688
LIFESTYLE            4.094904
MAPS_AND_NAVIGATION  4.051613
MEDICAL              4.189143
NEWS_AND_MAGAZINES   4.132189
PARENTING            4.300000
PERSONALIZATION      4.335256
PHOTOGRAPHY          4.192114
PRODUCTIVITY         4.211396
SHOPPING             4.259664
SOCIAL               4.255598
SPORTS               4.223511
TOOLS                4.047203
TRAVEL_AND_LOCAL     4.109292
VIDEO_PLAYERS        4.063750
WEATHER              4.244000

## Count Number of Apps in each Genre

In [29]:
df_genre = df.Genres.value_counts().reset_index().rename(columns = {"index": "Genres", "Genres": "Count"})
df_genre

Genres  Count
0                                  Tools    732
1                          Entertainment    533
2                              Education    468
3                                 Action    358
4                           Productivity    351
..                                   ...    ...
110                   Board;Pretend Play      1
111              Entertainment;Education      1
112                   Strategy;Education      1
113  Health & Fitness;Action & Adventure      1
114                     Card;Brain Games      1

[115 rows x 2 columns]

## Group App by Genre, Find Mean, Median & Mode Rating of each Genre

In [20]:
df.groupby("Genres").mean().sort_values("Rating",ascending = False)

Rating
Genres                                       
Comics;Creativity                    4.800000
Board;Pretend Play                   4.800000
Health & Fitness;Education           4.700000
Adventure;Brain Games                4.600000
Strategy;Action & Adventure          4.600000
...                                       ...
Art & Design;Pretend Play            3.900000
Health & Fitness;Action & Adventure  3.900000
Educational                          3.871875
Parenting;Education                  3.866667
Parenting;Brain Games                3.800000

[115 rows x 1 columns]

In [21]:
df.groupby("Genres").median().sort_values("Rating", ascending = False)

Rating
Genres                                     
Comics;Creativity                      4.80
Board;Pretend Play                     4.80
Health & Fitness;Education             4.70
Art & Design;Creativity                4.70
Puzzle;Education                       4.60
...                                     ...
Educational                            3.95
Art & Design;Pretend Play              3.90
Health & Fitness;Action & Adventure    3.90
Parenting;Education                    3.90
Parenting;Brain Games                  3.80

[115 rows x 1 columns]

In [22]:
df.groupby("Genres").agg(pd.Series.mode)

Category Rating  \
Genres                                                                  
Action                                                    GAME    4.3   
Action;Action & Adventure                               FAMILY    4.3   
Adventure                                                 GAME    4.4   
Adventure;Action & Adventure                            FAMILY    4.5   
Adventure;Brain Games                                   FAMILY    4.6   
...                                                        ...    ...   
Video Players & Editors                          VIDEO_PLAYERS    4.3   
Video Players & Editors;Creativity     [FAMILY, VIDEO_PLAYERS]    4.1   
Video Players & Editors;Music & Video                   FAMILY      4   
Weather                                                WEATHER    4.2   
Word                                                      GAME    4.7   

                                                            Size  \
Genres                                                             
Action                                        Varies with device   
Action;Action & Adventure                                    95M   
Adventure                                     Varies with device   
Adventure;Action & Adventure                                 67M   
Adventure;Brain Games                                       8.1M   
...                                                          ...   
Video Players & Editors                       Varies with device   
Video Players & Editors;Creativity                           23M   
Video Players & Editors;Music & Video         Varies with device   
Weather                                       Varies with device   
Word                                   [87M, Varies with device]   

                                                 Installs  Type  Price  \
Genres                                                                   
Action                                           10000000  Free      0   
Action;Action & Adventure                        10000000  Free      0   
Adventure                              [100000, 10000000]  Free      0   
Adventure;Action & Adventure                    100000000  Free      0   
Adventure;Brain Games                              100000  Paid  $2.99   
...                                                   ...   ...    ...   
Video Players & Editors                           1000000  Free      0   
Video Players & Editors;Creativity                5000000  Free      0   
Video Players & Editors;Music & Video            10000000  Free      0   
Weather                                           1000000  Free      0   
Word                                             10000000  Free      0   

                                      Content Rating        Last Updated  \
Genres                                                                     
Action                                          Teen      August 1, 2018   
Action;Action & Adventure                   Everyone       June 29, 2018   
Adventure                                       Teen       July 23, 2018   
Adventure;Action & Adventure            Everyone 10+       July 31, 2018   
Adventure;Brain Games                       Everyone  September 13, 2017   
...                                              ...                 ...   
Video Players & Editors                     Everyone      August 3, 2018   
Video Players & Editors;Creativity          Everyone       July 17, 2018   
Video Players & Editors;Music & Video   Everyone 10+       June 21, 2018   
Weather                                     Everyone      August 2, 2018   
Word                                        Everyone      August 2, 2018   

                                                        Current Ver  \
Genres                                                                
Action                                           Varies with device   
Action;Action & Adventure                             

In [23]:
df.groupby("Genres").describe()

Rating                                 \
                                       count      mean       std  min   25%   
Genres                                                                        
Action                                 358.0  4.285475  0.291353  3.1  4.20   
Action;Action & Adventure               17.0  4.311765  0.172780  3.9  4.30   
Adventure                               73.0  4.180822  0.312542  3.4  4.00   
Adventure;Action & Adventure            13.0  4.423077  0.148064  4.1  4.50   
Adventure;Brain Games                    1.0  4.600000       NaN  4.6  4.60   
...                                      ...       ...       ...  ...   ...   
Video Players & Editors                158.0  4.063924  0.554566  1.8  3.80   
Video Players & Editors;Creativity       2.0  4.100000  0.000000  4.1  4.10   
Video Players & Editors;Music & Video    3.0  4.000000  0.000000  4.0  4.00   
Weather                                 75.0  4.244000  0.331353  3.3  4.05   
Word                                    28.0  4.410714  0.325849  3.8  4.20   

                                                      
                                       50%  75%  max  
Genres                                                
Action                                 4.3  4.5  5.0  
Action;Action & Adventure              4.3  4.4  4.6  
Adventure                              4.3  4.4  4.7  
Adventure;Action & Adventure           4.5  4.5  4.5  
Adventure;Brain Games                  4.6  4.6  4.6  
...                                    ...  ...  ...  
Video Players & Editors                4.2  4.4  4.9  
Video Players & Editors;Creativity     4.1  4.1  4.1  
Video Players & Editors;Music & Video  4.0  4.0  4.0  
Weather                                4.3  4.5  4.8  
Word                                   4.4  4.7  4.9  

[115 rows x 8 columns]

## Insight

1. Dari tabel Googleplaystore, terlihat ada 33 kategori dan 115 genre yang berbeda. Tabel menunjukkan info-info tentang apps di Google Play Store seperti Rating, Installs, Price.


2. Apps yang paling mahal adalah I'm Rich - Trump Edition($400). Apps yang mengikutinya juga I'm Rich dengan variasi edisinya. Kategori lifestyle dan finance mendominasi apps termahal. 


3. Apps yang termurah ada banyak dengan harga $0.99. Apps yang murah lebih mengarah pada kategori personalization.


4. Kategori apps yang paling banyak di Play Store adalah kategori family dengan 1746 apps dibawah kategori tersebut. 


5. App yang paling banyak didownload adalah Viber Messenger (500juta+). Ini karena Viber memiliki video dan voice call quality yang bagus dan lebih aman untuk dipakai. Top 10 apps lebih mengarah pada sosial (communication, social, game) karena kategori tersebut cater kepada berbagai masyarakat.


6. Ketika mengkalkulasi average rating dari kategori, rating tertinggi adalah Arts and Design.


7. Genre yang punya paling banyak apps adalah genre Tools(732).


8. Rating mean dan median tertinggi adalah genre Comics